source/common/router/router.cc

```diff
@@ -2032,9 +2032,9 @@ bool Filter::checkDropOverload(Upstream::ThreadLocalCluster& cluster,
     if (config_.random_.bernoulli(cluster.dropOverload())) {
       ENVOY_STREAM_LOG(debug, "The request is dropped by DROP_OVERLOAD", *callbacks_);
       callbacks_->streamInfo().setResponseFlag(StreamInfo::ResponseFlag::DropOverLoad);
-      chargeUpstreamCode(Http::Code::ServiceUnavailable, nullptr, true);
+      chargeUpstreamCode(Http::Code::BadGateway, nullptr, true);
```

Comment:
According to the [RFC 503 definition](https://www.rfc-editor.org/rfc/rfc9110.html#name-503-service-unavailable) of 503:
```
... indicates that the server is currently unable to handle the request due to a temporary overload or scheduled maintenance, which will likely be alleviated after some delay.
```
which seems like a better fit, no?